src/librustc/traits/project.rs

```diff
@@ -1047,8 +1049,8 @@ fn confirm_fn_pointer_candidate<'cx, 'gcx, 'tcx>(
     fn_pointer_vtable: VtableFnPointerData<'tcx, PredicateObligation<'tcx>>)
     -> (Ty<'tcx>, Vec<PredicateObligation<'tcx>>)
 {
-    // FIXME(#32730) propagate obligations (fn pointer vtable nested obligations ONLY come from
-    // unification in inference)
+    // FIXME(#32730) drop this assertion once obligations are propagated from inference (fn pointer
+    // vtable nested obligations ONLY come from unification in inference)
     assert!(fn_pointer_vtable.nested.is_empty());
```

Comment:
note that in the  "thread vtable-closure obligations to result" commit of https://github.com/rust-lang/rust/pull/33816 I fixed a bug -- probably introduced by me -- where the comparable `vtable.nested` obligations were being totally ignored in the fn below